ethportal-api/src/types/content_key/state.rs

```diff
@@ -143,224 +157,117 @@ impl fmt::Display for StateContentKey {
-                sha256.finalize().into()
-            }
-        }
-    }
-
-    fn to_bytes(&self) -> Vec<u8> {
-        panic!("Not implemented: Implement in pr along w/ new state content key spec update.")
-    }
-}
-
 #[cfg(test)]
 #[allow(clippy::unwrap_used)]
 mod test {
+    use std::str::FromStr;
+
     use super::*;
     use crate::utils::bytes::hex_decode;
 
     //
     // State Network Content Key Tests
     //
 
-    const STATE_ROOT: [u8; 32] = [
-        0xd1, 0xc3, 0x90, 0x62, 0x4d, 0x3b, 0xd4, 0xe4, 0x09, 0xa6, 0x1a, 0x85, 0x8e, 0x5d, 0xcc,
-        0x55, 0x17, 0x72, 0x9a, 0x91, 0x70, 0xd0, 0x14, 0xa6, 0xc9, 0x65, 0x30, 0xd6, 0x4d, 0xd8,
-        0x62, 0x1d,
-    ];
+    const NIBBLES: [u8; 12] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
 
     const ADDRESS: [u8; 20] = [
-        0x82, 0x9b, 0xd8, 0x24, 0xb0, 0x16, 0x32, 0x6a, 0x40, 0x1d, 0x08, 0x3b, 0x33, 0xd0, 0x92,
-        0x29, 0x33, 0x33, 0xa8, 0x30,
+        0x00, 0x0d, 0x83, 0x62, 0x01, 0x31, 0x8e, 0xc6, 0x89, 0x9a, 0x67, 0x54, 0x06, 0x90, 0x38,
+        0x27, 0x80, 0x74, 0x32, 0x80,
     ];
 
-    const CODE_HASH: [u8; 32] = [
-        0xd1, 0xc3, 0x90, 0x62, 0x4d, 0x3b, 0xd4, 0xe4, 0x09, 0xa6, 0x1a, 0x85, 0x8e, 0x5d, 0xcc,
-        0x55, 0x17, 0x72, 0x9a, 0x91, 0x70, 0xd0, 0x14, 0xa6, 0xc9, 0x65, 0x30, 0xd6, 0x4d, 0xd8,
-        0x62, 0x1d,
+    const HODE_HASH: [u8; 32] = [
+        0xc5, 0xd2, 0x46, 0x01, 0x86, 0xf7, 0x23, 0x3c, 0x92, 0x7e, 0x7d, 0xb2, 0xdc, 0xc7, 0x03,
+        0xc0, 0xe5, 0x00, 0xb6, 0x53, 0xca, 0x82, 0x27, 0x3b, 0x7b, 0xfa, 0xd8, 0x04, 0x5d, 0x85,
+        0xa4, 0x70,
     ];
 
     #[test]
-    fn account_trie_node() {
-        let expected_content_key = "0x0044000000b8be7903aee73b8f6a59cd44a1f52c62148e1f376c0dfa1f5f773a98666efc2bd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d01020001";
-        let expected_content_id: [u8; 32] = [
-            0x5b, 0x2b, 0x5e, 0xa9, 0xa7, 0x38, 0x44, 0x91, 0x01, 0x0c, 0x1a, 0xa4, 0x59, 0xa0,
-            0xf9, 0x67, 0xdc, 0xf8, 0xb6, 0x99, 0x88, 0xad, 0xbf, 0xe7, 0xe0, 0xbe, 0xd5, 0x13,
-            0xe9, 0xbb, 0x83, 0x05,
-        ];
-
-        let node_hash = [
-            0xb8, 0xbe, 0x79, 0x03, 0xae, 0xe7, 0x3b, 0x8f, 0x6a, 0x59, 0xcd, 0x44, 0xa1, 0xf5,
-            0x2c, 0x62, 0x14, 0x8e, 0x1f, 0x37, 0x6c, 0x0d, 0xfa, 0x1f, 0x5f, 0x77, 0x3a, 0x98,
-            0x66, 0x6e, 0xfc, 0x2b,
-        ];
-        let path = vec![1, 2, 0, 1];
-        let node = AccountTrieNode {
-            node_hash,
-            state_root: STATE_ROOT,
-            path: path.into(),
-        };
-
-        let key = StateContentKey::AccountTrieNode(node);
-        let encoded: Vec<u8> = key.clone().into();
-
-        assert_eq!(hex_decode(expected_content_key).unwrap(), encoded);
-        assert_eq!(expected_content_id, key.content_id());
+    fn account_trie_node_key() {
+        let expected_content_key = "0x2024000000c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a4700005000000123456789abc";
```

Comment:
It seems appropriate to add these state test vectors to the [specs](https://github.com/ethereum/portal-network-specs/tree/master) now, in the same way that we do for history & beacon networks.

Reply:
There is already [github gist](https://gist.github.com/mynameisdaniil/9cb64fa3206b07e8c20cfa23178f5030) with test vectors. I took values from there.

The spec PR is in more or less final stage, but not yet merged. Once that one is in, I will create a PR to add test vectors as well.